python/hsfs/core/vector_server.py

```diff
@@ -264,13 +279,34 @@ def get_feature_vectors(self, entry, passed_features=[]):
             )
         )
 
-        return list(
+        # get column names for pandas dataframe
+        feature_names = list(
+            map(
+                lambda result_dict: self._generate_vector(result_dict)[1],
```

Comment:
you call `self._generate_vector(result_dict)` and in line 292. Can you instead do 
`generated_vector = self._generate_vector(result_dict)` at the beginning and re-use the object?

Reply:
good catch. done